lib/utils.ts

```diff
@@ -51,3 +51,31 @@ export const validateEmailWithACL = (email) => {
   }
   return false;
 };
+
+/**
+ * This sets `cookie` using the `res` object
+ */
+export const setCookie = (
+  res: NextApiResponse,
+  name: string,
+  value: unknown,
+  options: { path?: string } = {}
+) => {
+  const stringValue = typeof value === 'object' ? JSON.stringify(value) : String(value);
+  let cookieContents = name + '=' + stringValue;
+  if (options.path) {
+    cookieContents += '; Path=' + options.path;
+  }
+  res.setHeader('Set-Cookie', cookieContents);
+};
+
+export const JACKSON_ERROR_COOKIE_KEY = 'jackson_error';
+
+// returns the cookie with the given name,
+// or undefined if not found
+export function getCookie(name) {
```

Comment:
Looks good @niwsa. I would suggest we expose this as setErrorCookie and getErrorCookie and hide JACKSON_ERROR_COOKIE_KEY internally. Also if lib/utils is being called from frontend isn't it safer to separate out the cookie logic from utils in case the utils scope expands in the future to include server side code?